kornia/geometry/transform/image_registrator.py

```diff
@@ -0,0 +1,263 @@
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}({self.model})'
+
+    def reset_model(self):
+        """Initializes the model with identity transform."""
+        torch.nn.init.eye_(self.model)
+
+    def forward(self) -> torch.Tensor:
+        r"""Single-batch homography".
+
+        Return:
+            Homography.
+
+        Shape:
+            - Output: :math:`(1, 3, 3)`"""
+        return torch.unsqueeze(self.model / self.model[2, 2], dim=0)  # 1x3x3
+
+    def forward_inverse(self) -> torch.Tensor:
+        r"""Interted Single-batch homography".
+
+        Return:
+            Homography.
+
+        Shape:
+            - Output: :math:`(1, 3, 3)`"""
+        return torch.unsqueeze(torch.inverse(self.model), dim=0)
+
+
+class Similarity(nn.Module):
+    def __init__(self,
+                 rotation: bool = True,
+                 scale: bool = True,
+                 shift: bool = True):
+        """Similarity geometric model to be used together with ImageRegistrator module for the optimization-based
+        image registration.
+
+        Args:
+            rotation (bool): if True, the rotation is optimizable, else constant zero
+            scale (bool): if True, the scale is optimizable, else constant zero
+            shift (bool): if True, the shift is optimizable, else constant one
+        """
+        super().__init__()
+        if rotation:
+            self.rot = nn.Parameter(torch.Tensor(1))
+        else:
+            self.register_buffer('rot', torch.zeros(1))
+        if shift:
+            self.shift = nn.Parameter(torch.Tensor(1, 2, 1))
+        else:
+            self.register_buffer('shift', torch.zeros(1, 2, 1))
+        if scale:
+            self.scale = nn.Parameter(torch.Tensor(1))
+        else:
+            self.register_buffer('scale', torch.ones(1))
+        self.reset_model()
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}(angle = {self.rot},\
+               \n shift={self.shift}, \n scale={self.scale})'
+
+    def reset_model(self):
+        """Initializes the model with identity transform."""
+        torch.nn.init.zeros_(self.rot)
+        torch.nn.init.zeros_(self.shift)
+        torch.nn.init.ones_(self.scale)
+
+    def forward(self) -> torch.Tensor:
+        r"""Single-batch similarity transform".
+
+        Return:
+            Similarity.
+
+        Shape:
+            - Output: :math:`(1, 3, 3)`"""
+        rot = self.scale * angle_to_rotation_matrix(self.rot)
+        out = convert_affinematrix_to_homography(torch.cat([rot, self.shift],
+                                                           dim=2))
+        return out
+
+    def forward_inverse(self) -> torch.Tensor:
+        r"""Single-batch inverse similarity transform".
+
+        Return:
+            Similarity.
+
+        Shape:
+            - Output: :math:`(1, 3, 3)`"""
+        return torch.inverse(self.forward())
+
+
+class ImageRegistrator(nn.Module):
```

Comment:
add a small example snippet

Reply:
It is already there, see under "register"